pushmsixscript/src/pushmsixscript/microsoft_store.py

```diff
@@ -0,0 +1,151 @@
+        url = _store_url(config, f"{application_id}/submissions/{submission_to_remove}")
+        session.delete(url, headers=headers, timeout=config["request_timeout_seconds"])
+        _log_response(response)
+        response.raise_for_status()
+
+
+def _create_submission(config, channel, session, headers):
+    # create a new in-progress submission, which is a copy of the last published submission
+    application_id = config["application_ids"][channel]
+    url = _store_url(config, f"{application_id}/submissions")
+    response = session.post(url, headers=headers, timeout=config["request_timeout_seconds"])
+    _log_response(response)
+    response.raise_for_status()
+    return response.json()
+
+
+def _update_submission(config, channel, session, submission_request, headers, file_path):
+    # update the in-progress submission, including uploading the new msix file
+    application_id = config["application_ids"][channel]
+    submission_id = submission_request.get("id")
+    upload_url = submission_request.get("upload_url")
+    upload_url = upload_url.replace("+", "%2B")
+    url = _store_url(config, f"{application_id}/submissions/{submission_id}")
+    response = session.put(url, submission_request, headers=headers)
+    _log_response(response)
+    response.raise_for_status()
+    with open(file_path, "rb") as f:
+        upload_headers = {"x-ms-blob-type": "BlockBlob"}
+        response = requests.put(upload_url, f, headers=upload_headers, timeout=config["request_timeout_seconds"])
+        response.raise_for_status()
+        _log_response(response)
+        return response.json()
+
+
+def _commit_submission(config, channel, session, submission_id, headers):
+    # finalize submission
+    application_id = config["application_ids"][channel]
+    url = _store_url(config, f"{application_id}/submissions/{submission_id}/commit")
+    response = session.post(url, headers=headers, timeout=config["request_timeout_seconds"])
+    _log_response(response)
+    response.raise_for_status()
+    return response.json()
+
+
+def _get_submission_status(config, channel, session, submission_id, headers):
+    application_id = config["application_ids"][channel]
+    url = _store_url(config, f"{application_id}/submissions/{submission_id}/status")
+    response = session.get(url, headers=headers, timeout=config["request_timeout_seconds"])
+    _log_response(response)
+    response.raise_for_status()
+    return response.json()
+
+
+def _wait_for_commit_completion(config, channel, session, submission_id, headers):
+    # pull submission status until commit process is completed
+    response_json = _get_submission_status(config, channel, session, submission_id, headers)
+    attempts = 1
+    while response_json.get("status") == "CommitStarted":
+        if attempts > COMMIT_POLL_MAX_ATTEMPTS:
+            return False
+        attempts += 1
+        time.sleep(COMMIT_POLL_WAIT_SECONDS)
```

Comment:
Depending on how long we need to wait on this and how flaky it can be, we could go the `notarization_poller` route. Or maybe we just spin up enough pushmsix workers to deal :)

Reply:
Unfortunately, I don't have any sense of how long it will take yet, since this only happens once we actually commit a new submission. I'm looking into how / when I can do a fuller test.